src/devices/bus/spectrum/zxbus.cpp

```diff
@@ -41,12 +40,40 @@ zxbus_device::zxbus_device(const machine_config &mconfig, const char *tag, devic
 
 zxbus_device::zxbus_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
 	: device_t(mconfig, type, tag, owner, clock)
+	, device_memory_interface(mconfig, *this)
+	, m_io_config("io", ENDIANNESS_LITTLE, 8, 17, 0, address_map_constructor())
 	, m_iospace(*this, finder_base::DUMMY_TAG, -1)
+	, m_allocspaces(false)
+{
+}
+
+device_memory_interface::space_config_vector zxbus_device::memory_space_config() const
 {
+	return space_config_vector {
+		std::make_pair(AS_IO, &m_io_config),
+	};
+}
```

Comment:
If only want the space sometimes, you don’t want to return the configuration for it if you don’t want it.